### PR TITLE
added edgegateway, Development section in README

### PR DESCRIPTION
**Pull request by floriankammermann**
Added a HowTo develop seciton in the README

Added the possibility to list edgegateway. Was already possible over search, but the goal is to add more features to the edgegateway. The tests were done manually on our vCloud director.

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/vcd-cli/151)

### CODE REVIEW

File: vcd_cli/edgegateway.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            restore_session(ctx)
+            if not ctx.obj['profiles'].get('vdc_in_use') or \
+               not ctx.obj['profiles'].get('vdc_href'):
+                raise Exception('select a virtual datacenter')
+        except Exception as e:
+            stderr(e, ctx)
+
+@edgegateway.command('list', short_help='list EdgeGateways')
+@click.pass_context
+def list_edgegateways(ctx):
```
Comment (pacogomez) [inline vcd_cli/edgegateway.py]: the implementation should use the method in `pyvcloud` and not duplicate the code here.

**Comment by pacogomez** [comment]
see my comments inline

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/151#-:-L1TxDHp-G2xu4B-yWFh:bv668sj)*

**Comment by anusuyar** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[vcd_cli/edgegateway.py, line 33 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/151#-L1SDp7Vr9gqB-QNbd6Y-r1-33:-L1U0g7RJ9XfDFGovSYa:b4myad) ([raw file](https://github.com/vmware/vcd-cli/blob/89dce4061cf1e63c2cc1635219037f5c847f56b5/vcd_cli/edgegateway.py#L33)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

It should say `list of edge gateways` in the docstring.
While `edgegateway` is the right name, in the previous version of `vcd-cli` we had a shorter name for this command: `gateway`, for usability purposes (less to type). Other options could be `edge` or `gw`. @rocknes @anusuyar @sahithi what do you think? Also note that in the VDC representation there are two types of edges: `edgeGateways` and `fullEdgeGateways`.

</blockquote></details>

`vcd gateway list` seems better

---

*[vcd_cli/edgegateway.py, line 48 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/151#-L1U1ANVBQvLnpy1mhwU:-L1U1ANVBQvLnpy1mhwV:b-azbma3) ([raw file](https://github.com/vmware/vcd-cli/blob/89dce4061cf1e63c2cc1635219037f5c847f56b5/vcd_cli/edgegateway.py#L48)):*
> ```Python
>             stderr(e, ctx)
> 
> @edgegateway.command('list', short_help='list EdgeGateways')
> ```

list edge gateways

---

*[vcd_cli/edgegateway.py, line 50 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/151#-L1U1QZGLP-_2Ea081CF:-L1U1QZGLP-_2Ea081CG:b9xt2la) ([raw file](https://github.com/vmware/vcd-cli/blob/89dce4061cf1e63c2cc1635219037f5c847f56b5/vcd_cli/edgegateway.py#L50)):*
> ```Python
> @edgegateway.command('list', short_help='list EdgeGateways')
> @click.pass_context
> def list_edgegateways(ctx):
> ```

method name can be list_gateways

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/151)*

**Comment by pacogomez** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[vcd_cli/edgegateway.py, line 33 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/151#-L1SDp7Vr9gqB-QNbd6Y-r1-33:-L1tFXmFsFN4NpYSl4pI:b650qwc) ([raw file](https://github.com/vmware/vcd-cli/blob/89dce4061cf1e63c2cc1635219037f5c847f56b5/vcd_cli/edgegateway.py#L33)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

`vcd gateway list` seems better
</blockquote></details>

Florian, please use `gateway` as the name of the command.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/151)*

**Comment by pacogomez** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/151#-:-L2RHBOIALymmRqR4hXo:bnfp4nl)*

**Pull request merged**